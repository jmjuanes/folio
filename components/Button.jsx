import React from "react";
import classNames from "classnames";

export const Button = props => {
    const classList = classNames(props.className, {
        "flex items-center justify-center gap-2 select-none": true,
        "p-3 rounded-md": true,
        "cursor-pointer": !props.disabled,
        "o-70 cursor-not-allowed": props.disabled,
        "w-full": props.fullWidth,
    });
    return (
        <div data-testid={props.testid} className={classList} onClick={props.onClick} style={props.style}>
            {!!props.icon && (
                <div className={classNames(props.iconClassName, "flex items-center")} data-testid="btn-icon">
                    {props.icon}
                </div>
            )}
            {!!props.text && (
                <div className={props.textClassName} data-testid="btn-text">
                    <span>{props.text}</span>
                </div>
            )}
        </div>
    );
};

Button.defaultProps = {
    testid: "btn",
    className: "",
    text: "",
    textClassName: "text-sm",
    icon: null,
    iconClassName: "text-xl",
    disabled: false,
    fullWidth: false,
    style: null,
    onClick: null,
};

export const PrimaryButton = props => (
    <Button
        testid="btn-primary"
        className="bg-neutral-950 hover:bg-neutral-900 text-white"
        {...props}
    />
);

export const SecondaryButton = props => (
    <Button
        testid="btn-secondary"
        className="bg-white text-neutral-900 hover:bg-neutral-200 border border-neutral-200"
        {...props}
    />
);
