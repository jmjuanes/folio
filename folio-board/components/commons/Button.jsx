import React from "react";
import classNames from "classnames";

export const Button = props => {
    const classList = classNames(props.className, {
        "d:flex items:center justify:center gap:2 select:none": true,
        "p:3 r:lg": true,
        "cursor:pointer": !props.disabled,
        "o:70 cursor:not-allowed": props.disabled,
    });

    return (
        <div className={classList} onClick={props.onClick}>
            {!!props.icon && (
                <div className={classNames(props.iconClassName, "d:flex items:center")} data-test="icon">
                    {props.icon}
                </div>
            )}
            {!!props.text && (
                <div className={props.textClassName} data-test="text">
                    {props.text}
                </div>
            )}
        </div>
    );
};

Button.defaultProps = {
    className: "",
    text: "",
    textClassName: "",
    icon: null,
    iconClassName: "text:xl",
    disabled: false,
    onClick: null,
};
