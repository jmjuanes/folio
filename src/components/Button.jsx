import React from "react";
import classNames from "classnames";

export const Button = props => {
    const classList = classNames(props.className, {
        "flex items-center justify-center gap-2 select-none": true,
        "p-3 rounded-lg": true,
        "cursor-pointer": !props.disabled,
        "o-70 cursor-not-allowed": props.disabled,
        "w-full": props.fullWidth,
    });

    return (
        <div className={classList} onClick={props.onClick} style={props.style}>
            {!!props.icon && (
                <div className={classNames(props.iconClassName, "flex items-center")} data-test="icon">
                    {props.icon}
                </div>
            )}
            {!!props.text && (
                <div className={props.textClassName} data-test="text">
                    {props.text}
                </div>
            )}
        </div>
    );
};

Button.defaultProps = {
    className: "",
    text: "",
    textClassName: "",
    icon: null,
    iconClassName: "text-2xl",
    disabled: false,
    fullWidth: false,
    style: null,
    onClick: null,
};

export const PrimaryButton = props => (
    <Button
        className="bg-gray-800 hover:bg-gray-900 text-white"
        textClassName="text-sm"
        iconClassName="text-xl"
        {...props}
    />
);

export const SecondaryButton = props => (
    <Button
        className="bg-white hover:bg-gray-100 border border-gray-300"
        textClassName="text-sm"
        iconClassName="text-xl"
        {...props}
    />
);
