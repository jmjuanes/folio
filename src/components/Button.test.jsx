import TestRenderer from "react-test-renderer";
import {Button} from "./Button.jsx";

describe("Button", () => {
    it("should render", () => {
        const testRenderer = TestRenderer.create(<Button />);

        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    it("should render button text and icon", () => {
        const testRenderer = TestRenderer.create(<Button text="TEXT" icon="ICON" />);
        const testInstance = testRenderer.root;

        expect(testInstance.findByProps({"data-test":"icon"}).children[0]).toBe("ICON");
        expect(testInstance.findByProps({"data-test":"text"}).children[0]).toBe("TEXT");
    });
});
