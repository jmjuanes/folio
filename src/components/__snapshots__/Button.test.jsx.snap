// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<div
  className="flex items-center justify-center gap-2 select-none p-3 rounded-lg cursor-pointer"
  onClick={null}
/>
`;
